src/main/main.cpp

```diff
@@ -173,23 +288,58 @@ doMigrationStepOne(
             txHashes.push_back(ripple::uint256::fromVoid(buf));
         }
 
-        auto const txs =
-            doTryFetchTransactions(timer, backend, txHashes, yield);
+        auto txs = doTryFetchTransactions(timer, backend, txHashes, yield);
+        assert(txs.size() == txHashes.size());
```

Comment:
This assert will not do anything in release. Maybe it is better to crash here instead of writing wrong data?